kubernetes/README.md

```diff
@@ -44,7 +51,7 @@ The below example will create an autoscaled `Deployment` with 2 containers, a `S
 
   // Application deployment
   webServiceDeployment:
-    $.letsbuildServiceDeployment(
+    lbKubernetes.letsbuildServiceDeployment(
```

Comment:
Actually, I find the named version much more readable than `$`, so that could be a general pattern

Reply:
I'd say it depends. Here I changed it because the import is no longer a merge but a `local` assignment so it wouldn't work anymore. In principal, I agree. The named versions is more readable but I kind of think it would be an overkill to assign a local in some cases. Like here https://github.com/letsbuilders/letsbuild-kubernetes-clusters/blob/main/environments/monitoring/dev1/metrics/main.jsonnet#L33 I've decided it's better to just use `$.thanos` rather than have

```jsonnet
local thanos = $.thanos,

image: thanos._config.thanos.image,
objectStorageConfig: thanos._config.thanos.objectStorageConfig,
version: thanos._config.thanos.version,
```

My current modus operandi is to import anything external with locals. Like here I declared `local lbKubernetes`. The community is split about that and some people would just merge that with the root document but I've found it makes debugging a lot harder sometimes. But then again, if some modules are a part of a bigger coherent module then merging them makes a lot of sense - like here https://github.com/letsbuilders/accounts/blob/master/deploy/lib/letsbuilders/accounts.libsonnet

If you have a module that contains a couple `*.libsonnet` files, I'd rather have them all available under `$.`. Using locals in such a scenario can be sometimes confusing especially with bigger modules. 

I hope anything from that mumbo jumbo has made any sense. And, as always, I'm more than happy to discuss this further.